### PR TITLE
Extend CompositeDisposable init with varying number of disposables

### PR DESCRIPTION
**Pull request by larryonoff**

### CODE REVIEW

File: RxSwift/Disposables/CompositeDisposable.swift [hunk 1]
```diff
@@ -30,18 +30,10 @@ public class CompositeDisposable : DisposeBase, Disposable, Cancelable {
...
-    }
-    
-    /**
-     Initializes a new instance of composite disposable with the specified number of disposables.
-    */
-    public init(_ disposable1: Disposable, _ disposable2: Disposable, _ disposable3: Disposable) {
-        _disposables!.insert(disposable1)
-        _disposables!.insert(disposable2)
-        _disposables!.insert(disposable3)
+    public init(_ disposables: Disposable...) {
```
Comment (kzaher) [inline RxSwift/Disposables/CompositeDisposable.swift]: Hi @larryonoff ,

I am aware of variadic arguments, but those overloads where there as an optimization so there is no need to create a variadic argument array.

I see your point, and we are really missing a general version. 

Could you just add a version with 4 arguments + variadic rest to make sure we are using optimized version up to 3 arguments.

Comment (larryonoff) [inline RxSwift/Disposables/CompositeDisposable.swift]: Sure. I'll add it in an hour.

Comment (larryonoff) [inline RxSwift/Disposables/CompositeDisposable.swift]: Hope that I understand you properly, so I added 3 overloads.

File: RxSwift/Disposables/CompositeDisposable.swift [hunk 2]
```diff
@@ -47,6 +49,26 @@ public class CompositeDisposable : DisposeBase, Disposable, Cancelable {
     /**
      Initializes a new instance of composite disposable with the specified number of disposables.
     */
+    public init(_ disposable1: Disposable, _ disposable2: Disposable, _ disposable3: Disposable, _ disposable4: Disposable) {
```
Comment (kzaher) [inline RxSwift/Disposables/CompositeDisposable.swift]: Hi @larryonoff ,

I've meant to merge this one `public init(_ disposable1: Disposable, _ disposable2: Disposable, _ disposable3: Disposable, _ disposable4: Disposable)` with `public init(_ disposables: Disposable...) {`

into `public init(_ disposable1: Disposable, _ disposable2: Disposable, _ disposable3: Disposable, _ disposable4: Disposable, _ disposables: Disposable...)`

If you could squash those these commits and make a PR to `develop` branch, I would be happy to merge this.

**Pull request closed without merging**